### ISSUE TITLE
[Feature Request]: Conditional page break

### ISSUE THREAD

**Issue by pilotkid** [carboneio/carbone#91]
**Problem to solve**
I would like a way to add a page break programmatically. this way items can be formatted more efficiently.

**Proposed solution**
A formatter that adds a page break on condition. 
For example ```d.toBuy:ifEQ(true):pageBreak```

**Describe alternatives you've considered**
Adding a show if block based off of data containing a page break.
For example 
```
{d.toBuy:ifEQ(true):showBegin}
**---PAGE BREAK---**
{d.toBuy:ifEQ(true):showEnd}
```
The issue with this format is that you have to place the logic above whatever you format which can make it difficult to figure out what the report will look like without generating it. Also with a page break on the first page, it can be confusing why it is there and has caused team members to delete the page break. 

**Additional context**
**Currently**
![image](https://user-images.githubusercontent.com/6385285/89595354-88d62280-d811-11ea-842a-4ea94774231d.png)

**Proposed**
![image](https://user-images.githubusercontent.com/6385285/89595633-3fd29e00-d812-11ea-8b76-fa6f9b3b7e6a.png)

**Comment by steevepay** [comment]
Hello @pilotkid, thank you for the feature request!

Right now it is not possible to add a page break based on a conditional formatter. 
I am going to discuss your proposed solution with the team.

Until I come back to you, the following solution is available to create a page break:
```
{ d.array[i] }
*your content*
*PAGE_BREAK*
*LINE_BREAK*
{ d.array[i+1] }
```

**Comment by pilotkid** [comment]
@steevepay Yep that works the only issue is that it leaves an empty page at the end of the dataset, and setting `IsSkipEmptyPages` in the render loop, still renders the page. Probably due to it having a header and footer. 

Thank you for your response

**Comment by wahyubaskara** [comment]
The above solution it worked, but it's render a blank page at the end of the document. Any solution to remove the blank page at the end of document?

**Comment by BeaG130** [comment]
Hello,  

In case someone has the same question today, we wrote an article about this:

https://help.carbone.io/en-us/article/how-to-insert-a-page-break-without-having-a-blank-page-j6jy00/?bust=1757408332437
